server/onerror.ts

```diff
@@ -81,33 +82,43 @@ export default function onerror(app: Koa) {
       err = newError;
     }
 
+    this.status = err.status ?? 500;
+
     if (err instanceof ValidationError) {
-      // @ts-expect-error status is not a property on ValidationError
-      err.status = 400;
+      this.status = 400;
 
       if (err.errors && err.errors[0]) {
         err.message = `${err.errors[0].message} (${err.errors[0].path})`;
       }
     }
 
+    if (err instanceof formidable.errors.FormidableError) {
+      // Client aborted errors are a 500 by default, but 499 is more appropriate
+      if (err.internalCode === 1002) {
+        this.status = 499;
+      } else {
+        this.status = err.httpCode ?? 500;
+      }
+    }
+
     if (
       err.code === "ENOENT" ||
       err instanceof EmptyResultError ||
       /Not found/i.test(err.message)
     ) {
-      err.status = 404;
+      this.status = 404;
```

Comment:
These changes are probably causing test failures.

Reply:
Decided to pull the error changes into a separate PR, going to refactor that handler a bit